e2e/peer_ops_test.go

```diff
@@ -25,8 +27,13 @@ func TestAddRemovePeer(t *testing.T) {
 	r.True(g2.IsRunning())
 
 	// add peer: ask g1 to add g2 as peer
-	reqBody := strings.NewReader(fmt.Sprintf(`{"addresses": ["%s"]}`, g2.PeerAddress))
-	resp, err := http.Post("http://"+g1.ClientAddress+"/v1/peers", "application/json", reqBody)
+	peerAddReq := api.PeerAddReq{
+		Addresses: []string{g2.PeerAddress},
+	}
+	reqBody, err := json.Marshal(peerAddReq)
+	r.Nil(err)
+
+	resp, err := http.Post("http://"+g1.ClientAddress+"/v1/peers", "application/json", strings.NewReader(string(reqBody)))
```

Comment:
This patch is not meant to change the e2e behavior. Only changes related to pkg/api are changed here. I will send separate patch to fix e2e tests using restclient.